edb/schema/types.py

```diff
@@ -2108,6 +2133,370 @@ def get_underlying_schema_class(cls) -> typing.Type[Collection]:
+
+    def get_implicit_cast_distance(
+        self, other: Type, schema: s_schema.Schema
+    ) -> int:
+        if not isinstance(other, Range):
+            return -1
+
+        return self.get_element_type(schema).get_implicit_cast_distance(
+            other.get_element_type(schema), schema)
+
+    def assignment_castable_to(
+        self,
+        other: Type,
+        schema: s_schema.Schema,
+    ) -> bool:
+        if not isinstance(other, Range):
+            return False
+
+        return self.get_element_type(schema).assignment_castable_to(
+            other.get_element_type(schema), schema)
+
+    def castable_to(
+        self,
+        other: Type,
+        schema: s_schema.Schema,
+    ) -> bool:
+        if not isinstance(other, Range):
+            return False
+
+        return self.get_element_type(schema).castable_to(
+            other.get_element_type(schema), schema)
+
+    def find_common_implicitly_castable_type(
+        self: Range_T,
+        other: Type,
+        schema: s_schema.Schema,
+    ) -> typing.Tuple[s_schema.Schema, Optional[Range_T]]:
+
+        if not isinstance(other, Range):
+            return schema, None
+
+        if self == other:
+            return schema, self
+
+        my_el = self.get_element_type(schema)
+        schema, subtype = my_el.find_common_implicitly_castable_type(
+            other.get_element_type(schema), schema)
+
+        if subtype is None:
+            return schema, None
+
+        return type(self).from_subtypes(schema, [subtype])
+
+    def _resolve_polymorphic(
+        self,
+        schema: s_schema.Schema,
+        concrete_type: Type,
+    ) -> Optional[Type]:
+        if not isinstance(concrete_type, Range):
+            return None
+
+        return self.get_element_type(schema).resolve_polymorphic(
+            schema, concrete_type.get_element_type(schema))
+
+    def _to_nonpolymorphic(
+        self,
+        schema: s_schema.Schema,
+        concrete_type: Type,
+    ) -> typing.Tuple[s_schema.Schema, Range]:
+        return Range.from_subtypes(schema, (concrete_type,))
+
+    def _test_polymorphic(self, schema: s_schema.Schema, other: Type) -> bool:
+        if other.is_any(schema):
+            return True
+
+        if not isinstance(other, Range):
+            return False
+
+        return self.get_element_type(schema).test_polymorphic(
+            schema, other.get_element_type(schema))
+
+    @classmethod
+    def from_subtypes(
+        cls: typing.Type[Range_T],
+        schema: s_schema.Schema,
+        subtypes: Sequence[Type],
+        typemods: Any = None,
+        *,
+        name: Optional[s_name.QualName] = None,
+        **kwargs: Any,
+    ) -> typing.Tuple[s_schema.Schema, Range_T]:
+        if len(subtypes) != 1:
+            raise errors.SchemaError(
+                f'unexpected number of subtypes, expecting 1: {subtypes!r}')
+        stype = subtypes[0]
+
+        # XXX: need to make sure only anypoint is allowed here
```

Comment:
What's stopping you from implementing this check?

Reply:
Implemented